pages/tidal tools/vsphere.md

```diff
@@ -6,62 +6,87 @@ sidebar: main_sidebar
+6. Skip HTTPS certificate checks? - The deafult is to not skip (n), which can be left unless you know they need to be skipped.
 
-After which, a set of prompts will appear to help you connect to your vSphere account:
+After answering the above prompts, you will see the message
 
-1. vSphere Server
-2. vSphere Username
-3. vSphere Password
-4. vSphere API Endpoint
-5. Use SSL/TLS?
-6. Skip HTTPS certificate checks?
+```
+Login to vSphere successful. Saving config file...Done!
+```
 
-After answering the above prompts, you will see the message _"Login to vSphere successful. Saving config file...Done!"_
+## Sync {#vsphere-sync}
+Syncronizing your vSphere inventory with Tidal Migrations is simple with:
 
+`tidal sync vsphere`
 
-## Configure
-Once you have Tidal Tools installed you need to configure access to vSphere and Tidal Migrations.
+```
+Loading data..
+Transforming data..
+Validating JSON...
+Valid.
+Logging in to server at https://demo.tidalmg.com as info@tidalmigrations.com
+Logged in.
+Updating via API..
+Update successful
+```
 
-The credentials can be defined using (1) the tidal config command, (2) with a config file or (3) via environment variables. If a config file is present it will be used, if it is not present environment variables will be used. If none of these are defined, you will be prompted to enter the required details.
+You can easily split this into two commands to allow you to inject other metadata about your inventory, if needed.
+To do this simply use the get command to get your inventory.
+Modify it as you need and then you can use the sync command to upload your data.
 
+1. `tidal get vsphere`, will fetch your vSphere inventory and output JSON (check it out!).
 
+2. Modify the data as you like from Standard Input and then send it to Standard Output.
 
-### 1) Set interactively
+3. `tidal sync servers`, will take that JSON from STDIN and send it to Tidal Migrations.
 
-You can set your vSphere credentials with the following commands:
+```
+tidal get vsphere | ./modify_script.rb | tidal sync servers
+```
 
-``` tidal config set vsphere.username [your username] ```
+## Repeat
+You can easily set this to run periodically. The integration updates records if they already exist, or creates new records if they don't.
+Look at [setting this command up as a cron job](https://www.digitalocean.com/community/tutorials/how-to-use-cron-to-automate-tasks-on-a-vps) that once per day.
 
-``` tidal config set vsphere.password [your password] ```
+## Additional Configuration Options
+The integration can be configured with several other methods besides `tidal login`.
+You can use these alternate approaches if they better suite your use case.
 
-``` tidal config set vsphere.server 192.168.1.12 ```
+The other options are to use, the `tidal config` command, a config file via `--config` or via environment variables.
+If a config file is present it will be used, if it is not present environment variables will be used.
```

Comment:
I think AWS cli sets a good example of (how to do this)[https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-quickstart.html#cli-configure-quickstart-precedence]

The Tidal Tools credentials and configuration settings take precedence in the following order:
1. Configuration file
2. Environment Variables
3. Prompt in the terminal

Reply:
Good idea. I followed that pattern here too. I took it a bit farther and added some more to make it clear too. Also, I think the precedence is '1. env var / 2. config file', so I listed it that way.

What do you think of how this reads now?

Reply:
Yeah it reads much nice now! Yeah, I was just following the order that you had in the sentences - so it was meant more as a rough idea than something to adhere to.